src/dbnode/storage/database.go

```diff
@@ -708,8 +716,31 @@ func (d *db) QueryIDs(
 	query index.Query,
 	opts index.QueryOptions,
 ) (index.QueryResult, error) {
+	var (
+		sp    opentracing.Span
+		spCtx stdctx.Context
+	)
+
+	goCtx, exists := ctx.GoContext()
+	if exists {
+		sp, spCtx = opentracingutil.StartSpanFromContext(goCtx, dbQuerySpanID)
```

Comment:
Does this span automatically get tied together with the span from the RPC layer?

Also could we add a method to our context that makes adding each span less redundant? Maybe something like:

```golang
sp, closer := ctx.StartSpan(
    spanID,
    opentracinglog.String("query", query.String()),
    opentracinglog.String("namespace", namespace.String()),
    opentracinglog.Int("limit", opts.Limit),
    opentracinglog.Time("start", opts.StartInclusive),
    opentracinglog.Time("start", opts.StartInclusive),
)
defer closer()
sp.LogFields()
...
```

and all of the `ctx.GoContext()` `if exists` etc boilerplate can get shoved into that helper method

That way its a little cleaner/shorter and the API allows you to add the logs that you have immediately right away and then if you need to add more later you can use the returned `sp` object.